### PR TITLE
add `createProgram` docs

### PR DESCRIPTION
**Pull request by donaldpipowitch**
Refers to a little bit to #15693. I try to learn the compiler API and have a "hard" time to find good documentation and resources. I noticed I can't use code completion to _see_ the documentation for certain APIs and I thought it would be a good idea to add them.

I took documentation from https://github.com/Microsoft/TypeScript/wiki/Architectural-Overview,  https://github.com/Microsoft/TypeScript/wiki/Using-the-Language-Service-API and https://github.com/Microsoft/TypeScript/wiki/Using-the-Compiler-API to write this comment.

Are you interested in pull requests like that?

### CODE REVIEW

File: src/compiler/program.ts [hunk 1]
```diff
@@ -290,6 +290,15 @@ namespace ts {
         return resolutions;
     }
 
+    /**
+     * Parses a set of root files in order, walks their dependency declaration (both imports and triple-slash references)
```
Comment (mhegazy) [inline src/compiler/program.ts]: how about?
```txt
Create a new 'Program' instance. A Program is an immutable collection of 'SourceFiles' and a compilation options that represent a compilation unit. 

Creating a program proceeds from a set of rootFiles, expanding the set of inputs by following imports and triple-slash reference directives transitively.  `@types` and triple-slash-reference-types are also pulled in.
```

**Comment by donaldpipowitch** [comment]
Updated :)

**Comment by mhegazy** [comment]
thanks @donaldpipowitch!

**Comment by donaldpipowitch** [comment]
Thank you :)

**Pull request merged**